lib/iris/tests/__init__.py

```diff
@@ -695,93 +702,99 @@ def _assert_graphic(self):
             result_fname = os.path.join(image_output_directory,
                                         'result-' + unique_id + '.png')
 
-            if not os.path.isdir(os.path.dirname(result_fname)):
+            if not os.path.isdir(image_output_directory):
                 # Handle race-condition where the directories are
                 # created sometime between the check above and the
                 # creation attempt below.
                 try:
-                    os.makedirs(os.path.dirname(result_fname))
+                    os.makedirs(image_output_directory)
                 except OSError as err:
                     # Don't care about "File exists"
                     if err.errno != 17:
                         raise
 
-            def _save_figure_hash():
-                plt.gcf().savefig(result_fname)
-                # Determine the test result image hash using sha1.
-                with open(result_fname, 'rb') as fi:
-                    sha1 = hashlib.sha1(fi.read())
-                return sha1
-
             def _create_missing():
-                fname = sha1.hexdigest() + '.png'
-                base_uri = ('https://scitools.github.io/test-images-scitools/'
-                            'image_files/{}')
+                fname = '{}.png'.format(phash)
+                base_uri = ('https://scitools.github.io/test-iris-imagehash/'
+                            'images/{}')
                 uri = base_uri.format(fname)
                 hash_fname = os.path.join(image_output_directory, fname)
                 uris = repo.setdefault(unique_id, [])
                 uris.append(uri)
                 print('Creating image file: {}'.format(hash_fname))
-                os.rename(result_fname, hash_fname)
+                figure.savefig(hash_fname)
                 msg = 'Creating imagerepo entry: {} -> {}'
                 print(msg.format(unique_id, uri))
                 with open(repo_fname, 'wb') as fo:
                     json.dump(repo, codecs.getwriter('utf-8')(fo), indent=4,
                               sort_keys=True)
 
-            sha1 = _save_figure_hash()
+            # TBD: Push this fix to imagehash (done!)
+            # See https://github.com/JohannesBuchner/imagehash/pull/31
+            # Now need this imagehash/master pushed to pypi ...
+            def _hex_to_hash(hexstr, hash_size=_HASH_SIZE):
+                l = []
+                count = hash_size * (hash_size // 4)
+                if len(hexstr) != count:
+                    emsg = 'Expected hex string size of {}.'
+                    raise ValueError(emsg.format(count))
+                for i in range(count // 2):
+                    h = hexstr[i*2:i*2+2]
+                    v = int("0x" + h, 16)
+                    l.append([v & 2**i > 0 for i in range(8)])
+                return imagehash.ImageHash(np.array(l))
```

Comment:
This fix has already been merged to `imagehash` but the fix has not yet been pushed to `pypi` ... we can remove this function when the fix is available via `pypi` .... or even conda-forge!